Extensions/Physics2Behavior/physics2runtimebehavior.ts

```diff
@@ -1014,13 +1019,18 @@ namespace gdjs {
         this._objectOldY !== this.owner.getY() ||
         this._objectOldAngle !== this.owner.getAngle()
       ) {
-        const pos = this.b2Vec2(
+        const x =
           (this.owner.getDrawableX() + this.owner.getWidth() / 2) *
-            this._sharedData.worldInvScale,
+          this._sharedData.worldInvScale;
+        const y =
           (this.owner.getDrawableY() + this.owner.getHeight() / 2) *
-            this._sharedData.worldInvScale
+          this._sharedData.worldInvScale;
+        const pos = this.b2Vec2(
+          Number.isFinite(x) ? x : body.GetPosition().x,
+          Number.isFinite(y) ? y : body.GetPosition().y
         );
-        body.SetTransform(pos, gdjs.toRad(this.owner.getAngle()));
+        const angle = gdjs.toRad(this.owner.getAngle());
+        body.SetTransform(pos, Number.isFinite(angle) ? x : body.GetAngle());
```

Comment:
is this line right? should this be `body.SetTransform(pos, Number.isFinite(angle) ? angle : body.GetAngle());` ? (angle instead of x)

Also, just to check the `body.getAngle()` is already a Rad value right?

Reply:
Yes, Box2D use radiant.